newlib/libc/stdlib/random.c

```diff
@@ -55,14 +55,14 @@ algorithm as <<rand>>.
 #define _DEFAULT_SOURCE
 #include <stdlib.h>
 
-__THREAD_LOCAL long long _rand_next = 1;
+__THREAD_LOCAL unsigned long long _rand_next = 1;
```

Comment:
This looks good, but we should make this actually match what's in taocp and use uint64_t instead of 'unsigned long long', which might not be 64 bits.